compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -2557,12 +2557,15 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         debug!(?predicates);
         assert_eq!(predicates.0.parent, None);
         let mut obligations = Vec::with_capacity(predicates.0.predicates.len());
-        for (predicate, span) in predicates.0.predicates {
+        for (impl_def_predicate_index, (predicate, span)) in
```

Comment:
tiny nit: This is a bit of an opinion, but `impl_def_predicate_index` doesn't really have *much* value here. A shorter name like `idx` is probably fine, it's particularly self-explanatory due to its use in a field with the same name.